boxtree/fmm.py

```diff
@@ -274,7 +276,10 @@ def finalize_potentials(self, potentials):
 
 
 def drive_fmm(wrangler: ExpansionWranglerInterface, src_weight_vecs,
-        timing_data=None):
+              timing_data=None, comm=None,
```

Comment:
Could we hide the communicator in a wrangler subclass? (Where communication is a no-op in the base case, and subclasses add that actual functionality? Then `_communicate_mpoles_via_allreduce` could just be another wrangler subclass instead of a flag here.)

Reply:
Implemented in https://github.com/inducer/boxtree/pull/8/commits/7112bdf6c851cab27856c4ea5d77f25352ab00e1.